app/controllers/answers_controller.rb

```diff
@@ -37,6 +37,15 @@ def update
       return update_as_suggested_edit
     end
 
+    if params[:answer][:body_markdown] == @answer.body_markdown
+      same_body = true
+    end
+
+    if same_body
+      flash[:danger] = 'No changes were saved, there were none to save.'
+      return redirect_to question_path(@answer.parent)
+    end
+
```

Comment:
You can merge these two blocks:

```suggestion
    if params[:answer][:body_markdown] == @answer.body_markdown
      flash[:danger] = "No changes were saved because you didn't edit the post."
      return redirect_to question_path(@answer.parent)
    end
```